Fix remote profile

luksa opened the pull request:
**Please provide a description of this PR:**

- Fix inconsistent usages of `global.externalIstiod` (this setting should only be used in the primary cluster and should not affect remote clusters at all)

- Modify the charts so that the `istiod` service is always called `istiod`, even in remote clusters. This simplifies things and prevents bugs, such as referring to the wrong service name in webhooks. 

- Modify the charts so that `.Values.istiodRemote.enabled` determines whether  `istio-reader-clusterrole` gives access to configmaps and webhooks.

istio-testing commented:
Hi @luksa. Thanks for your PR.

I'm waiting for a [istio](https://github.com/orgs/istio/people) member to verify that this patch is reasonable to test. If it is, they should reply with `/ok-to-test` on its own line. Until that is done, I will not automatically test new commits in this PR, but the usual testing commands by org members will still work. Regular contributors should [join the org](https://github.com/orgs/istio/people) to skip this step.

Once the patch is verified, the new status will be reflected by the `ok-to-test` label.

I understand the commands that are listed [here](https://go.k8s.io/bot-commands?repo=istio%2Fistio).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository.
</details>

keithmattix commented:
/ok-to-test

bleggett reviewed (commented):
Yep, as mentioned in https://github.com/istio/istio/pull/53294 - this flag - `externalIstiod` - is broken.

The purpose of the flag in docs and comments is claimed to be "set this on the cluster *exposing* the remote control plane".

Clearly (by virtue of this PR existing), that is *not* how the flag is actually used.

Additionally, if we are always setting it to true for the remote profile, it should simply be removed - the flag has no purpose outside of remote deployments.

If we can always set it to `true` in this profile, then we can clearly simply remove it and replace relevant usages of it with `istiodRemote.enabled = true`.

So - rather than encode a useless flag in the profile, just remove `externalIstiod` flag, it's pointless if `istiodRemote.enabled` is true, which it will be in every case where this profile is used.

luksa commented:
@bleggett Yeah, I had a feeling that was the case - I was confused why the flag was used on both sides. I'll remove the flag. What should we do with the Service name? IMHO it should be `istiod-remote` so that it's immediately clear that it points to an external control plane. Of course, we also need to update the reference to the service in the webhook. You agree?

luksa commented:
Hmm, maybe it's safer to always use `istiod` as the service name. If we use a different name based on whether istiod is local or not, we'll always have to keep this in mind whenever we reference the service by name (e.g. in the mutating webhooks).

bleggett commented:
> @bleggett Yeah, I had a feeling that was the case - I was confused why the flag was used on both sides. I'll remove the flag. What should we do with the Service name? IMHO it should be `istiod-remote` so that it's immediately clear that it points to an external control plane. Of course, we also need to update the reference to the service in the webhook. You agree?

I think we have tests that expect the service naming to go both ways (`-remote` suffix vs no), and I think that's all the flag really controls, meaningfully.

I have no preference, I would pick whichever one requires the least integ/doctest churn.

luksa committed:
Fix remote profile

- Fix inconsistent usages of global.externalIstiod (this setting should only be used in the primary cluster and should not affect remote clusters at all)

- Modify the charts so that the `istiod` service is always called `istiod`, even in remote clusters. This simplifies things and prevents bugs, such as referring to the wrong service name in webhooks.

- Modify the charts so that `.Values.istiodRemote.enabled` determines whether  `istio-reader-clusterrole` gives access to configmaps and webhooks.

istio-testing commented:
@luksa: The following test **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
integ-distroless_istio | 5fcfd49afb1a7e2178f15bf51b666a6ce650d5fe | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53542/integ-distroless_istio/1846533248809701376) | true | `/test integ-distroless`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

luksa commented:
/retest

MorrisLaw commented on operator/cmd/mesh/manifest-generate_test.go:
```diff
@@ -381,14 +381,15 @@ func TestManifestGenerateIstiodRemote(t *testing.T) {
 		g.Expect(objs.kind(gvk.CustomResourceDefinition.Kind).nameEquals("adapters.config.istio.io")).Should(BeNil())
 		g.Expect(objs.kind(gvk.CustomResourceDefinition.Kind).nameEquals("authorizationpolicies.security.istio.io")).Should(Not(BeNil()))
 
+		const istiodServiceName = "istiod"
```
nit: the constant could be defined outside of the for loop; we shouldn't need to update the value for every iteration

luksa replied:
Done. Thanks.

luksa committed:
Move constant declaration out of loop

bleggett reviewed (approved):
Awesome ty - `externalIstiod` is now only used to set the pilot env now, so it's less overloaded with random behaviors.

istio-testing commented:
@luksa: The following test **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
unit-tests-arm64_istio | eb2e707bafe77d8ba3f36ccbf117299f9bd381be | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/53542/unit-tests-arm64_istio/1846619842149355520) | unknown | `/test unit-tests-arm64`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

jacob-delgado commented:
/retest
